### PR TITLE
Prevent duplicate team names fixes #52

### LINKED ISSUE #52: Prevent teams with duplicate names

### PR DESCRIPTION
**Pull request by melissajstudent**
The objective of this change was to prevent duplicate team names. Although the system can manage this (because each team has a unique identifier in the database), it makes it easier for users to track teams when each team displays a unique name.

The implementation of code to check for duplicate, regardless of case, was simple given the Rails built-in uniqueness helper. The challenge was managing the error.

The original code had an if statement in the model controller that, if there was an error, just presented a generic message (e.g. "Error creating team. Please try again."). This message is not useful to the user.

I looked up solutions and was introduced to I18n which allows for use of locales based on model, attribute, error type, etc. It's extremely flexible. However, I felt it was too involved for the task at hand and I need to implement something quickly to stay on course with Bloc requirements and implement now while many users are signing up for the new season. 

I decided to add the message as an attribute of the uniqueness helper. I then updated page view code to take advantage of the custom message and remove the display of error message from the controller.

I think this is the best approach because it allows developers to think about the custom message for each validation type as part of the model. I imagine I18n is more powerful in that by manipulating it, one could change the error message format (the way it is currently implemented will always display the model attribute name) and see all error messaging in one spot (the en.yml file). I may explore this under a separate issue/PR later.

### CODE REVIEW

File: app/views/home/index.html.erb [hunk 1]
```diff
@@ -7,7 +7,7 @@
       <strong>You can now pay online!</strong> We've implemented payment support through PayPal. If you'd like to pay online, click the Pay Now button and a new tab will open that allows you to either pay via a credit card or your PayPal account. Please note, the payment will appear as Upside Down Studios on your statement and you will be charged $10.61. The extra $0.61 is a fee that PayPal charges for transaction processing. You can still pay in person. See the FAQ for details.
     </div> -->
 
-	  <% if current_user.admin? || current_user.manager? %>
+	   <% if current_user.admin? || current_user.manager? %>
```
Comment (cyle-dawson-bloc) [inline app/views/home/index.html.erb]: Random space.

File: app/views/teams/new.html.erb [hunk 2]
```diff
@@ -1,12 +1,25 @@
...
+        <%= message %>
+       <% end %>
+     </div>
+  <% end %>
+
+  <div class = 'form-group'>
+    <br />
+    <p>Enter a name for your team. Your team name is used to identify you in the standings. Most just use their own name (e.g. 'John Doe') and if they have multiple teams, add '1', '2', etc. (e.g. 'John Doe 1', 'John Doe 2') but you can be as creative as you like.</p>
+    <%= f.text_field :name, class: 'form-control', placeholder: 'Team Name' %>
       <br />
```
Comment (cyle-dawson-bloc) [inline app/views/teams/new.html.erb]: Indented one too far.

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -1,5 +1,5 @@
-require 'codeclimate-test-reporter'
-CodeClimate::TestReporter.start
+# require 'codeclimate-test-reporter'
+# CodeClimate::TestReporter.start
```
Comment (cyle-dawson-bloc) [inline spec/spec_helper.rb]: Unless you know you are going to need it later, you should avoid leaving commented out code. If you will need it later, make a note above the code explaining that.

File: app/models/team.rb [hunk 4]
```diff
@@ -1,10 +1,10 @@
 class Team < ApplicationRecord
   belongs_to :user
   has_many :picks, dependent: :destroy
-  validates :name, presence: true
+  validates :name, presence: true, uniqueness: {case_sensitive: false, message: "has already been taken. Please use a different name."}
 
   after_create :send_new_team_email
-
+
```
[the remark this answers is not in the document]
Comment (cyle-dawson-bloc) [reply app/models/team.rb]: Missed this space from last time.

File: app/views/teams/edit.html.erb [hunk 5]
```diff
@@ -2,6 +2,17 @@
...
+	<% if @team.errors.any? %>
+		<div class = "alert alert-danger alert-dismissible" >
+			<button type="button" class="close" data-dismiss="alert" aria-label="Close">
+			    <span aria-hidden="true">&times;</span>
+			</button>
+			<% @team.errors.full_messages.each do |message| %>
+				<%= message %>
+			<% end %>
+		</div>
+	<% end %>
```
[the remark this answers is not in the document]
Comment (cyle-dawson-bloc) [reply app/views/teams/edit.html.erb]: Missed these tabs from last time.

File: app/views/home/index.html.erb (hunk 1 above)
Comment (melissajstudent) [reply app/views/home/index.html.erb]: Note to self - don't switch text editors!

**Comment by melissajstudent** [comment]
@cyle-dawson-bloc this should be fixed now.

**Pull request merged**